Generate a list of known PCIs for xorg to auto-detect them and use openchrome for those
Index: xserver-xorg-video-openchrome/src/Makefile.am
===================================================================
--- xserver-xorg-video-openchrome.orig/src/Makefile.am
+++ xserver-xorg-video-openchrome/src/Makefile.am
@@ -92,6 +92,12 @@ EXTRA_DIST += \
 endif
 
 
+pcidatadir = $(datadir)/xserver-xorg/pci
+pcidata_DATA = openchrome.ids
+
+openchrome.ids : via_id.h
+	awk '/^#define.*PCI_CHIP/ {print $$3}' ${srcdir}/via_id.h | sed -e 's/0x/1106/' > openchrome.ids
+
 via_driver.lo: svnversion.h
 svnversion.h: $(openchrome_drv_la_SOURCES)
 	@if [ -d .svn ]; then \
